components/eventing-controller/reconciler/subscription/reconciler.go

```diff
@@ -0,0 +1,964 @@
+		if err := r.updateCondition(subscription, condition, ctx); err != nil {
+			return statusChanged, err
+		}
+	}
+	// OK
+	return statusChanged, nil
+}
+
+// deleteBEBSubscription deletes the BEB subscription and updates the condition and k8s events
+func (r *Reconciler) deleteBEBSubscription(subscription *eventingv1alpha1.Subscription, logger logr.Logger, ctx context.Context) error {
+	logger.Info("Deleting BEB subscription")
+	if err := r.bebClient.DeleteBebSubscription(subscription); err != nil {
+		return err
+	}
+	condition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionSubscribed, eventingv1alpha1.ConditionReasonSubscriptionDeleted, corev1.ConditionFalse)
+	return r.updateCondition(subscription, condition, ctx)
+}
+
+func (r *Reconciler) syncAPIRule(subscription *eventingv1alpha1.Subscription, ctx context.Context, logger logr.Logger) (*apigatewayv1alpha1.APIRule, error) {
+	if subscription.DeletionTimestamp != nil {
+		logger.Info("subscription is getting deleted so nothing needs to be done")
+		return nil, nil
+	}
+
+	isValidSink, err := r.isSinkURLValid(ctx, subscription, logger)
+	if err != nil {
+		logger.Error(err, "failed to validate sink URLs")
+		return nil, err
+	}
+	if !isValidSink {
+		logger.Error(fmt.Errorf("sink URL is not valid"), subscription.Spec.Sink)
+		return nil, nil
+	}
+
+	sURL, err := url.ParseRequestURI(subscription.Spec.Sink)
+	if err != nil {
+		logger.Error(err, "failed to parse sink URI")
+		r.eventWarn(subscription, reasonValidationFailed, "Failed to parse sink URI %s", subscription.Spec.Sink)
+		return nil, nil
+	}
+	apiRule, err := r.createOrUpdateAPIRule(subscription, ctx, *sURL, logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to createOrUpdateAPIRule")
+	}
+	return apiRule, nil
+}
+
+func (r *Reconciler) isSinkURLValid(ctx context.Context, subscription *eventingv1alpha1.Subscription, logger logr.Logger) (bool, error) {
+	if !isValidScheme(subscription.Spec.Sink) {
+		r.eventWarn(subscription, reasonValidationFailed, "Sink URL scheme should be 'http' or 'https' %s", subscription.Spec.Sink)
+		return false, nil
+	}
+
+	sURL, err := url.ParseRequestURI(subscription.Spec.Sink)
+	if err != nil {
+		logger.Error(err, subscription.Spec.Sink)
+		r.eventWarn(subscription, reasonValidationFailed, "Sink URL is not valid %s", err.Error())
+		return false, nil
+	}
+
+	// Validate sink URL is a cluster local URL
+	trimmedHost := strings.Split(sURL.Host, ":")[0]
+	if !strings.HasSuffix(trimmedHost, clusterLocalURLSuffix) {
+		logger.Error(fmt.Errorf("sink does not contain suffix: %s in the URL", clusterLocalURLSuffix), "")
+		r.eventWarn(subscription, reasonValidationFailed, "sink does not contain suffix: %s in the URL", clusterLocalURLSuffix)
+		return false, nil
+	}
+	subDomains := strings.Split(trimmedHost, ".")
+	if len(subDomains) != 5 {
```

Comment:
(nit): A small comment could help understand the magic 5 :)

```suggestion
	# we expected a sink in the format "service.namespace.svc.cluster.local"
	if len(subDomains) != 5 {
```

Reply:
Done.